pkgs/flox-tests/default.nix

```diff
@@ -79,78 +81,47 @@ in
     EOF
         }
 
-        # Transform long options to short ones
-        for arg in "$@"; do
-          shift
-          case "$arg" in
-            '--flox')   set -- "$@" '-F';;
-            '--tests')  set -- "$@" '-T';;
-            '--watch')  set -- "$@" '-W';;
-            '--help')   set -- "$@" '-h';;
-            *)          set -- "$@" "$arg" ;;
+        WATCH=
+        while [[ "$#" -gt 0 ]]; do
+          case "$1" in
+            -[fF]|--flox)         export FLOX_CLI="''${2?}"; shift; ;;
+            -[tT]|--tests)        export TESTS_DIR="''${2?}"; shift; ;;
+            -[wW]|--watch)        WATCH=:; ;;
+            -h|--help|-u|--usage) usage; exit 0; ;;
+            --)                   shift; break; ;;
+            *)  echo "ERROR: Unrecognized arg '$*'" >&2; usage; exit 1; ;;
           esac
+          shift;
         done
 
-        WATCH=0
-
-        while getopts ":F:T:Wh" flag;
-        do
-          case "$flag" in
-            F)
-              export FLOX_CLI=$OPTARG
-              ;;
-            T)
-              export TESTS_DIR=$OPTARG
-              ;;
-            W)
-              WATCH=1
-              ;;
-            h)
-              usage
-              exit 0
-              ;;
-          esac
-        done
-
-        # remove options from positional parameters
-        shift $(expr $OPTIND - 1)
-
         # Default flag values
-        if [ -z "$FLOX_CLI" ];
-        then
-          export FLOX_CLI=$TMP_FLOX_CLI
-        fi
-        if [ -z "$TESTS_DIR" ];
-        then
-          export TESTS_DIR=$PWD/tests
-        fi
-
-        # DEBUG
-        #echo "WATCH: $WATCH"
-        #echo "FLOX_CLI: $FLOX_CLI"
-        #echo "TESTS_DIR: $TESTS_DIR"
-        #echo "BATS ARGS: $@"
+        : "''${FLOX_CLI:=$TMP_FLOX_CLI}";
+        : "''${TESTS_DIR:=$PWD/tests}";
+        export TESTS_DIR FLOX_CLI;
 
         # isolate git config
-        ssh-keygen -t ed25519 -q -N "" -f $TEMP_FLOX/id_ed25519
-        git config --global gpg.format ssh
-        git config --global user.signingkey $TEMP_FLOX/id_ed25519.pub
+        ssh-keygen -t ed25519 -q -N "" -f "$TEMP_FLOX/id_ed25519";
+        # FIXME: Why on earth is the test suite modifying my GPG perms?!
```

Comment:
Yeah, this should be moved into the setup part of the test suit (without the `--global` or isolate `git` better)